packages/hub/node-tests/indexers-test.js

```diff
@@ -88,6 +90,46 @@ describe('hub/indexers', function() {
     });
   });
 
+
+  describe('binary data', function() {
+    before(async function() {
+      let factory = new JSONAPIFactory();
+
+      factory.addResource('content-types', 'cardstack-files')
+        .withRelated('fields', [
+          factory.addResource('fields', 'created-at').withAttributes({fieldType: '@cardstack/core-types::date'}),
+          factory.addResource('fields', 'content-type').withAttributes({fieldType: '@cardstack/core-types::string'}),
+          factory.addResource('fields', 'sha-sum').withAttributes({fieldType: '@cardstack/core-types::string'}),
+          factory.addResource('fields', 'file-name').withAttributes({fieldType: '@cardstack/core-types::string'}),
+          factory.addResource('fields', 'size').withAttributes({fieldType: '@cardstack/core-types::integer'})
+        ]);
```

Comment:
I think you will get the schema for cardstack-files automatically if you make sure `@cardstack/files` is a dependency of stub-project.

Reply:
do you mean putting it in cardstack/packages/hub/node-tests/stub-middleware-extra/package.json? I tried that and it didn't work.

Reply:
No, I mean `${__dirname}/../../../tests/stub-project/package.json`. That's why `createDefaultEnvironment` takes a path as an argument, it needs to know where to find the package.json to know what plugins to use.

Reply:
ok I tried that _and_ added a `@cardstack/files` data source in `bootstrap-schema.js`, but the `default` ephemeral data source isn't created until you call `createDefaultEnvironment`. So it's sort of a Catch-22.